app/controllers/api/v1/data/historical_emissions_controller.rb

```diff
@@ -4,6 +4,7 @@ module Data
       class HistoricalEmissionsController < Api::V1::Data::ApiController
         include Streamable
         before_action :parametrise_filter, only: [:index, :download]
+        before_action :parametrise_metadata_filter, only: [:download]
```

Comment:
Basically I used a filter because there already was one filter in there and I thought it's best to match existing code. Looking at it now I realise it is too long and most of the logic it contains should go inside the filter class. I don't have an issue with filters in general, because they are a Rails classic that most devs are used to, so they're not really surprising. But filters with that much logic in them are a smell, so I'll try to address that.